app/models/api_key.rb

```diff
@@ -42,4 +59,8 @@ def other_enabled_scopes?
   def scope_presence
     errors.add :base, "Please enable at least one scope" unless enabled_scopes.any?
   end
+
+  def not_soft_deleted?
+    errors.add :base, "Key is invalid. Please delete and create a new one." if soft_deleted?
```

Comment:
This is different from the error message in `BaseController`, is that intentional?

Reply:
It was intentional as I thought the message in `BaseController` wouldn't make sense here. On second thought however, it does fit - I'll make both messages the same 👍